src/controller/LottoController.js

```diff
@@ -0,0 +1,124 @@
+
+    this.#completeLotto(lottoMoney, rankCounter);
+  }
+
+  async #prepareLotto() {
+    const { lottoMoney, userLotto } = await this.#generateUserLotto();
+    const { winningLottoNumbers, winningLottoBonusNumber } = await this.#generateWinningLotto();
+
+    return {
+      lottoMoney,
+      userLotto,
+      winningLottoNumbers,
+      winningLottoBonusNumber,
+    };
+  }
+
+  #playLotto(userLotto, winningLottoNumbers, winningLottoBonusNumber) {
+    const rankCounter = new RankCounter();
+    userLotto.forEach((lotto) => {
+      const rank = new Rank(
+        winningLottoNumbers.getNumbers(),
+        winningLottoBonusNumber.getBonusNumber(),
+        lotto,
+      );
+
+      rankCounter.increaseRankCounter(rank.getRank());
+    });
+    return rankCounter.getRankCounterArray();
+  }
+
+  #completeLotto(lottoMoney, rankCounter) {
+    const profit = new Profit(lottoMoney.getMoney(), rankCounter);
+    OutputView.printLottoStatics(rankCounter, profit.getProfit());
+  }
+
+  async #generateUserLotto() {
+    const lottoMoney = await this.#generateMoney();
+    const lottoCount = this.#generateLottoCount(lottoMoney);
+    const userLotto = this.#generateRandomLotto(lottoCount);
+    return { lottoMoney, userLotto };
+  }
+
+  async #generateWinningLotto() {
+    const winningLottoNumbers = await this.#generateWinningLottoNumbers();
+    const winningLottoBonusNumber =
+      await this.#generateWinningLottoBonusNumber(winningLottoNumbers);
+    return { winningLottoNumbers, winningLottoBonusNumber };
+  }
+
+  async #retryOnErrorTemplate(callback) {
+    while (true) {
+      try {
+        return await callback();
+      } catch (error) {
+        OutputView.printErrorMessage(error.message);
+      }
+    }
+  }
+
+  async #generateMoney() {
+    return this.#retryOnErrorTemplate(async () => {
+      const inputMoney = await InputView.readMoney();
+      return new Money(inputMoney);
+    });
+  }
+
+  async #generateWinningLottoNumbers() {
+    return this.#retryOnErrorTemplate(async () => {
+      const winningNumbers = await InputView.readWinningNumbers();
+      return new Lotto(winningNumbers.split(','));
+    });
+  }
+
+  async #generateWinningLottoBonusNumber(winningLotto) {
+    return this.#retryOnErrorTemplate(async () => {
+      const bonusNumber = await InputView.readBonusNumber();
+      return new BonusLotto(winningLotto.getNumbers(), bonusNumber);
+    });
+  }
+
+  #generateLottoCount(lottoMoney) {
+    const lottoCount = lottoMoney.getMoney() / MONEY_UNIT;
+    OutputView.printLottoCounter(lottoCount);
+    return lottoCount;
+  }
+
+  #generateRandomLotto(lottoCounter) {
+    const lottoList = [];
+    for (let i = 0; i < lottoCounter; i++) {
+      const lotto = new Lotto(this.#selectRandomNumber());
+      lottoList.push(lotto.getNumbers());
+    }
+
+    OutputView.printLottoNumbers(lottoList);
+    return lottoList;
+  }
```

Comment:
`Array.from` 을 이용하면 좀 더 선언적인 코드가 될 것 같아요!

```suggestion
  #generateRandomLotto(lottoCounter) {
    const lottos = Array.from({length: lottoCounter}, () => new Lotto(this.#selectRandomNumber());
    const lottoList = lottos.map((lotto) => lotto.getNumbers());

    OutputView.printLottoNumbers(lottoList);
    return lottoList;
  }
```

Reply:
오..! 명령형 프로그래밍이 아닌 선언형 프로그래밍 방식을 제시해주신거군요?
`선언적 코드`에 대해 찾아보며, 개념에 대해 자세히 알 수 있었습니다.. 👍👍
선언적 코드는 `어떻게`보다 `무엇을`에 집중하는 것이군요~!

생각하지 못했던 조언 감사합니다! ㅎㅎ